crates/build/src/metadata.rs

```diff
@@ -199,6 +185,32 @@ pub(crate) fn execute(
     Ok(())
 }
 
+pub fn write_metadata(
```

Comment:
I think this would be more intuitive as an inherent method on `ContractMetadata`, since we are dealing with an instance of that in both cases.

Reply:
Yes and no. This function present a very specific functionality by **overwriting** the metadata.

Reply:
Okay then call it `overwrite_metadata`. From an API perspective it is more idiomatic to have operations which operate on an instance to be methods instead of free functions e.g. instead of:

```rust
crate::metadata::write_metadata(
            metadata_artifacts,
            metadata,
            build_steps,
            verbosity,
        )?;
```

have

```rust
metadata.write(
            metadata_artifacts,
            build_steps,
            verbosity,
        )?;
```

Reply:
This method is simply an extract of a snippet from `crate::medata::execute` and contains some printing log operations unsuitable for `metadata.write` idiomatically IMHO.

Reply:
It is a matter of opinion, I think operating on an instance where appropriate for code sharing is more readable and discoverable. But up to you in the end.